src/Microsoft.Identity.Client/Internal/ClientCredentialWrapper.cs

```diff
@@ -17,41 +24,154 @@ namespace Microsoft.Identity.Client.Internal
     /// </summary>
     internal sealed class ClientCredentialWrapper
     {
-        /// <summary>
-        /// Constructor of client (application) credentials from a <see cref="ClientAssertionCertificateWrapper"/>
-        /// </summary>
-        /// <param name="certificate">contains information about the certificate previously shared with AAD at application
-        /// registration to prove the identity of the application (the client) requesting the tokens.</param>
-        public ClientCredentialWrapper(ClientAssertionCertificateWrapper certificate)
+        public ClientCredentialWrapper(ApplicationConfiguration config)
         {
             ConfidentialClientApplication.GuardMobileFrameworks();
-            Certificate = certificate;
+
+            ValidateCredentialParameters(config);
+
+            switch (AuthenticationType)
```

Comment:
Since we have 4 different client credential authentication scenarios I added an enum to keep track of which one is selected to eliminate ambiguity.